src/DateTimePicker.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import cx from 'classnames';
+import Flatpickr from './FlatpickrComponent';
+import TimeInput from './TimeInput';
+import DateTimeLocalInput from './DateTimeLocalInput';
+
+/**
+ * @module DateTimePicker
+ */
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		super(props);
+
+		const propsValue = props.value || {};
+		this.state = {
+			value: {
+				date: propsValue.date || '',
+				time: propsValue.time || ''
+			},
+			isDateTimeLocalSupported: false
+		};
+
+		this.setDate = this.setDate.bind(this);
+		this.setTime = this.setTime.bind(this);
+		this.parseDateTime = this.parseDateTime.bind(this);
+	}
+
+	componentWillMount() {
+		this.setState({ isDateTimeLocalSupported: this.hasBrowserSupport() });
```

Comment:
`componentWillMount` [will be called on the server](https://facebook.github.io/react/docs/react-component.html#componentwillmount) where `document` will be undefined.

Move this to `componentDidMount`